src/vm/amd64/unixasmmacros.inc

```diff
@@ -52,7 +52,6 @@ C_FUNC(\Name):
 #endif
 C_FUNC(\Name):
         .cfi_startproc
-        .cfi_def_cfa_offset 8
```

Comment:
.cfi_def_cfa_offset 8 was before the symbol label originally. The stack unwinding happened to work with it, but it was causing problems on Mac.

The Mac problem was fixed by moving it after the label in #372 . It broke stack unwinding when stopped at the very beginning of the method in gdb - it should repro for you on the live tree right now. lldb works fine - it has generally more robust stack unwinding.

This change makes our assembly closer to what C++ compiler generates. It works because .cfi_startproc seems to imply cfa_register=rsp and cfa_offset=8.

I was considering redoing the macros so that our code is exactly like what C++ generates, but decided against it because of it did not looked like a net improvement. @janvorli - thoughts?